Makefile

```diff
@@ -103,7 +103,7 @@ check: fmt vet lint generate shadow exhaustive deadcode goimports govulncheck ##
 # ==================================================================================== #
 unit-test: ## Run unit tests
 	@echo "==> Running unit tests..."
-	ENABLE_INTEGRATION_TESTS=false go test -v ./...
+	ENABLE_INTEGRATION_TESTS=false go test -v -race -cover -coverprofile=c.out ./...
```

Comment:
Running the tests with coverage is significantly slower, so I wouldn't make it the default option in your makefile.

Reply:
Perhaps a dedicated makefile function that includes coverage would be better

Reply:
Got it. I will add a separate option fo it